src/planner.c

```diff
@@ -726,6 +726,18 @@ valid_hook_call(void)
 	return ts_extension_is_loaded() && planner_hcache_exists();
 }
 
+static bool
+dml_involves_hypertable(PlannerInfo *root, Hypertable *ht, Index rti)
```

Comment:
does this still throw an error if the hypertable delete is inside a CTE. ie `WITH cte as (DELETE FROM compressed_ht RETURNING *) SELECT * FROM CTE` I ask because we get the resultRelation from the root and I'm not sure if thats the root of the query or the cte. In any case I don't think we have a test for that, but should.

Reply:
root will be the root of the cte when planning the subquery, i've added a test